src/spend.rs

```diff
@@ -497,6 +502,40 @@ fn derived_desc(
     desc.derive(coin.deriv_index, secp)
 }
 
+/// Get value to use for transaction nLockTime in order to
+/// discourage fee sniping.
+///
+/// The approach follows that taken by Bitcoin Core:
+/// - most of the time, the value returned will be the current
+/// block height, but will randomly be up to 100 blocks earlier.
```

Comment:
If anyone's wondering why we set the current block height and not the next block height (which is the one we want to be the first valid), it's because by Bitcoin consensus `nLockTime` is not the first height at which this transaction is valid but the last height at which it is invalid.

Reply:
I was wondering this when first starting on the issue :slightly_smiling_face: I should have added this comment in the code.